src/logic/azext_logic/manual/custom.py

```diff
@@ -0,0 +1,35 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+
+
+def logic_workflow_update(cmd,
+                          client,
+                          resource_group_name,
+                          name,
+                          definition=None,
+                          tags=None,
+                          state=None):
+
+    if definition is not None and "definition" not in definition:
+        raise CLIError(str(definition) + " does not contain a 'definition' key")
```

Comment:
There should be a specific error type instead of CLIError.

Reply:
raise ValidationError instead